test/KisSpec.hs

```diff
@@ -42,17 +44,37 @@ spec = do
                 pat <- req (CreatePatient $ Patient "xy")
                 void $ req (PlacePatient pat (toSqlKey 1)))
             `shouldThrow` (== ConstraintViolation)
-    describe "withSqliteKis" $ do
+    describe "withSqliteKis" $
          it "can run two clients in sequence" $
             withSqliteKis InMemory kisConfig $ \kis -> do
                 patId <- runClient kis $ req (CreatePatient $ Patient "Simon")
                 pats <- runClient kis $ req GetPatients
                 pats `shouldBe` [Entity patId (Patient "Simon")]
+    describe "withSqlitePool" $ do
+          it "can run two clients in sequence" $
+             withTempFile "/tmp/" "tmpKisDB" $ \fp _ ->
+                 withSqliteKis (PoolBackendType (T.pack fp) 10) kisConfig $ \kis ->
+                     do patId <- runClient kis $ req (CreatePatient $ Patient "Simon")
+                        pats <- runClient kis $ req GetPatients
+                        liftIO $ pats `shouldBe` [Entity patId (Patient "Simon")]
+          it "can run two clients in parallel" $
```

Comment:
`\req -> notify wakeUp req >> handleKisRequest backend req` Das ist die richtige Stelle. Hier ist auch notification vor dem eigentlichen Request. D.h. der Thread wacht schon auf, bevor überhaupt was in der DB steht oder?